### PR TITLE
Fix container.net.* metrics accuracy and tagging

### PR DESCRIPTION
**Pull request by vboulineau**
### What does this PR do?

Fix `container.net.*` metrics accuracy on Linux.

Currently `container.net.*` metrics are always emitted with high cardinality tags while the values may not represent actual container-level values but POD-level value (multiple containers in a POD) or host-level value (containers running in host network). With this bugfix, the `container.net.*` metrics won't be emitted for containers running in host network and a single timeserie will be emitted by POD when running multiple containers.
Finally, in non-Kubernetes environment, if multiple containers share the same network namespace, `container.net.*` metrics won't be emitted.

### Motivation

Fix `container.net.*` metrics accuracy

### Additional Notes

Identifying containers belonging to the same network namespace is done by checking `proc/<pid>/net/dev` inode value. As this file reports values per network namespace (and not per PID), it's actually the same for all processes in the same network namespace.
To identify "host network" use case, it assumes host PID1 to be in root network namespace.

### Possible Drawbacks / Trade-offs

Inaccurate metrics will stop being emitted. `docker` metrics are not affected as there was a dedicated behaviour for Docker check in the past.

### Describe how to test/QA your changes

Run the Agent on Kubernetes with PODs:
- Running single container: should see high cardinality tags on `container.net*` metrics
- Running multiple containers: should see POD tags on `container.net*` metrics
- Running host network: `container.net*` metrics not emitted.

### CODE REVIEW

File: releasenotes/notes/fix-container-net-3feaf380bed1a51d.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+fixes:
+  - |
+    Fix `container.net.*` metrics accuracy on Linux. Currently `container.net.*` metrics are always emitted with high cardinality tags while the values may not represent actual container-level values but POD-level value (multiple containers in a POD) or host-level value (containers running in host network). With this bugfix, the `container.net.*` metrics won't be emitted for containers running in host network and a single timeserie will be emitted by POD when running multiple containers. Finally, in non-Kubernetes environment, if multiple containers share the same network namespace, `container.net.*` metrics won't be emitted.
```
Comment (buraizu) [inline releasenotes/notes/fix-container-net-3feaf380bed1a51d.yaml:4]:
```suggestion
    Fix `container.net.*` metrics accuracy on Linux. Currently `container.net.*` metrics are always emitted with high cardinality tags while the values may not represent actual container-level values but POD-level values (multiple containers in a pod) or host-level values (containers running in host network). With this bug fix, the `container.net.*` metrics aren't emitted for containers running in host network and a single timeseries is emitted by pods when running multiple containers. Finally, in non-Kubernetes environments, if multiple containers share the same network namespace, `container.net.*` metrics won't be emitted.
```

**Review by buraizu — APPROVED** [review]
Thanks, approving with some minor updates suggested for the release note

**Review by hush-hush — APPROVED** [review]

**Review by L3n41c — APPROVED** [review]

File: pkg/util/system/namespace_linux_test.go [hunk 2]
```diff
@@ -17,26 +17,13 @@ import (
...
-	assert.NotNil(t, pid2HostNet)
-	assert.True(t, *pid2HostNet)
-
-	pid3inode, err := GetProcessNamespaceInode(fakeProc, "3", "net")
-	assert.NoError(t, err)
-
-	pid3HostNet := IsProcessHostNetwork(fakeProc, pid3inode)
-	assert.NotNil(t, pid3HostNet)
-	assert.False(t, *pid3HostNet)
 }
```
[the remark this answers is not in the document]
Comment (vboulineau) [reply pkg/util/system/namespace_linux_test.go:29]: No in this file, but yes added to `network_linux_test.go`

**Pull request merged**